File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
from collections import Counter as ctr
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        c = ctr(arr)
        return len(set(c.values())) == len(set(c.keys()))
        # return len(c.values()) == len(set(c.values()))

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Set<Integer> s = new HashSet<>();
        int[] f = new int[3000];
        for(int i : arr)
            ++f[i+1000];
        for(int i : f)
            if(i != 0)
                if(!s.add(i))
                    return false;
        return true;
    }
}